spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java

```diff
@@ -214,9 +212,21 @@ public void setAdviceChain(Advice[] adviceChain) {
 	 * @param recoveryInterval The recovery interval.
 	 */
 	public void setRecoveryInterval(long recoveryInterval) {
-		this.recoveryInterval = recoveryInterval;
+		this.recoveryBackOff = new FixedBackOff(recoveryInterval, FixedBackOff.UNLIMITED_ATTEMPTS);
+	}
+
+	/**
+	 * Specify the {@link BackOff} for interval between recovery attempts.
+	 * The default is 5000 ms, that is, 5 seconds.
```

Comment:
OK. Agreed. It would be more friendly and independent of the target implementation.
Although I guess @snicoll got those `5 sec` for `FixedBackOff` impl from the JMS `DefaultMessageListenerContainer` :smile: .
And moreover I see that JMS container continues to use the `DEFAULT_RECOVERY_INTERVAL` const.
Will fix together with Docs.